test/e2e/security_context.go

```diff
@@ -0,0 +1,214 @@
+		},
+		Spec: api.PodSpec{
+			SecurityContext: &api.PodSecurityContext{},
+			Containers: []api.Container{
+				{
+					Name:  "test-container",
+					Image: "gcr.io/google_containers/busybox",
+				},
+			},
+			RestartPolicy: api.RestartPolicyNever,
+		},
+	}
+
+	return pod
+}
+
+var _ = Describe("[Skipped] Security Context", func() {
+	framework := NewFramework("security-context")
+
+	It("should support pod.Spec.SecurityContext.RunAsUser", func() {
+		podName := "security-context-" + string(util.NewUUID())
+		pod := &api.Pod{
+			ObjectMeta: api.ObjectMeta{
+				Name:   podName,
+				Labels: map[string]string{"name": podName},
+			},
+			Spec: api.PodSpec{
+				SecurityContext: &api.PodSecurityContext{},
+				Containers: []api.Container{
+					{
+						Name:    "test-container",
+						Image:   "gcr.io/google_containers/busybox",
+						Command: []string{"sh", "-c", "id -u"},
+					},
+				},
+				RestartPolicy: api.RestartPolicyNever,
+			},
+		}
+
+		var uid int64 = 1001
+		pod.Spec.SecurityContext.RunAsUser = &uid
+
+		framework.TestContainerOutput("pod.Spec.SecurityContext.RunAsUser", pod, 0, []string{
+			fmt.Sprintf("%v", uid),
+		})
+	})
+
+	It("should support container.SecurityContext.RunAsUser", func() {
+		podName := "security-context-" + string(util.NewUUID())
+		pod := &api.Pod{
+			ObjectMeta: api.ObjectMeta{
+				Name:   podName,
+				Labels: map[string]string{"name": podName},
+			},
+			Spec: api.PodSpec{
+				SecurityContext: &api.PodSecurityContext{},
+				Containers: []api.Container{
+					{
+						Name:            "test-container",
+						Image:           "gcr.io/google_containers/busybox",
+						Command:         []string{"sh", "-c", "id -u"},
+						SecurityContext: &api.SecurityContext{},
+					},
+				},
+				RestartPolicy: api.RestartPolicyNever,
+			},
+		}
+
+		var uid int64 = 1001
+		var overrideUid int64 = 1002
+		pod.Spec.SecurityContext.RunAsUser = &uid
+		pod.Spec.Containers[0].SecurityContext.RunAsUser = &overrideUid
+
+		framework.TestContainerOutput("pod.Spec.SecurityContext.RunAsUser", pod, 0, []string{
+			fmt.Sprintf("%v", overrideUid),
+		})
+	})
+
+	It("should support volume SELinux relabling", func() {
+		// Write and read a file with an empty_dir volume
+		// with a pod with the MCS label s0:c0,c1
+		pod := getSecurityContextTestPod()
+		volumeName := "test-volume"
+		mountPath := "/mounted_volume"
+		pod.Spec.Containers[0].VolumeMounts = []api.VolumeMount{
+			{
+				Name:      volumeName,
+				MountPath: mountPath,
+			},
+		}
+		pod.Spec.Volumes = []api.Volume{
+			{
+				Name: volumeName,
+				VolumeSource: api.VolumeSource{
+					EmptyDir: &api.EmptyDirVolumeSource{
+						Medium: api.StorageMediumDefault,
+					},
+				},
+			},
+		}
+		pod.Spec.SecurityContext.SELinuxOptions = &api.SELinuxOptions{
+			Level: "s0:c0,c1",
+		}
+		pod.Spec.Containers[0].Command = []string{"sleep", "6000"}
+
+		client := framework.Client.Pods(framework.Namespace.Name)
+		defer client.Delete(pod.Name, nil)
```

Comment:
I'd move this to below the line that ensures there isn't an error creating the pod (just after 150)